#include "username.h"
#include "ui_username.h"
#include "password.h"
#include <QMessageBox>


Username::Username(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::Username)
{
    ui->setupUi(this);

}

Username::~Username()
{
    delete ui;
}

void Username::on_ForgotPassword_clicked()
{
    /*Password passwort;
    passwort.setModal(true);
    this->close();
    passwort.exec();*/

}

void Username::on_ButtonCheck_clicked()
{
    QString email = ui -> Email->text();
    QString safety1 = ui ->lES1->text();
    QString safety2 = ui -> lES2->text();


    //Daten werden eingelesen

    NProfil *NP = NProfil::getInstance();

    QString username = NP->ProfilNutzernameVergessen(email, safety1, safety2);
    if(username != "")
    {

        QMessageBox::information(this, "Username vergessen", "Deine Daten werden 端berpr端ft...");
        ui->lEUsername->setText(username);
        //testet, ob E-mail, S1 und S2 passt und gibt Username aus
    }
    //Felder leer
    else if(email == "" || safety1 == "" || safety2 == "")
    {
        QMessageBox::warning(this, "Username vergessen", "Bitte Benutzerdaten eingeben");


    }
   else
    {
       QMessageBox::warning(this, "Username vergessen", "Deine Email stimmt nicht mit deinen Daten 端berein.");
       //Fehlermeldung, wenn  E-mail, S1 und S2 nicht 端bereinstimmen
    }
}


void Username::on_pushButton_clicked()
{
    this->close();
}
